platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/command/ProfileInstallCommand.java

```diff
@@ -0,0 +1,257 @@
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.application.service.command;
+
+import static org.boon.Boon.fromJson;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Paths;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import java.util.stream.Collectors;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.karaf.bundle.core.BundleService;
+import org.apache.karaf.features.Dependency;
+import org.apache.karaf.features.Feature;
+import org.apache.karaf.features.FeaturesService;
+import org.apache.karaf.shell.api.action.Argument;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.codice.ddf.admin.application.service.ApplicationService;
+import org.codice.ddf.admin.application.service.ApplicationServiceException;
+import org.codice.ddf.security.common.Security;
+import org.osgi.framework.BundleException;
+import org.osgi.service.resolver.ResolutionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.security.Subject;
+import ddf.security.SubjectUtils;
+
+@Command(scope = "profile", name = "install", description = "Installs the profile")
+@Service
+public class ProfileInstallCommand extends AbstractProfileCommand {
+    @Argument(index = 0, name = "profileName", description = "Name of install profile to use", required = true, multiValued = false)
+    String profileName;
+
+    private Security security = Security.getInstance();
+
+    private static final String ADVANCED_PROFILE_INSTALL_FEATURES = "install-features";
+    private static final String ADVANCED_PROFILE_UNINSTALL_FEATURES = "uninstall-features";
+    private static final String ADVANCED_PROFILE_START_APPS = "start-apps";
+    private static final String ADVANCED_PROFILE_STOP_BUNDLES = "stop-bundles";
+    private static final EnumSet NO_AUTO_REFRESH = EnumSet.of(FeaturesService.Option.NoAutoRefreshBundles);
+    private static final String PROFILE_PREFIX = "profile-";
+    private static final String PROFILE_EXTENSION = ".json";
+    private static final String RESTART_WARNING = "Problems occurred during installing process. The system may be in an unknown state. It is recommended to install from scratch.";
```

Comment:
✏️  _"An unexpected error occurred during the installation process. The system is in unknown state. It is strongly recommended to restart the installation from the beginning."_?